std/functional.d

```diff
@@ -260,6 +208,133 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by `argc`. If no explicit
+        `argc` is given, it is inferred to be `paramNames.length`.)
+    $(LI Explicit names for the function parameters may be given as
+        `paramNames`. If more than `argc` names are supplied, the
+        extras will be ignored.)
+    $(LI The letters (`a` through `z`) are used as default names
+        for up to the first 26 function parameters.)
+    $(LI `args[index]` can always be used to access any parameter by index,
+        regardless of whether it has been given a name.))
+
+If `fun` is not a `string`,  `stringLambda` aliases itself to `fun`.
+In such cases, `stringLambda` does $(B not) verify that `fun` accepts
+the specified number of arguments. (Try $(XREF traits, arity) instead.)
+*/
+template stringLambda(alias fun, size_t argc, paramNames...)
+    if (paramNames.length == 0 || allSatisfy!(isSomeString, typeof(paramNames)))
```

Comment:
> What constraint seems bad to you `argc <= 100` or `argc >= paramNames.length`?

`argc <= 100` is a pointless arbitrary limitation on the parameter count, and a violation of the [zero, one, infinity rule](http://c2.com/cgi/wiki?ZeroOneInfinityRule).

`argc >= paramNames.length` adds unnecessary friction when trying to supply parameter names via an `AliasSeq` that may be reused across functions with different `argc`.